gateway/src/node/client_handling/websocket/connection_handler/eth_events.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright 2021 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
+
+use bip39::core::str::FromStr;
+use bip39::Mnemonic;
+use cosmrs::{Coin, Decimal};
+use rand::seq::SliceRandom;
+use rand::thread_rng;
+use url::Url;
+use web3::contract::tokens::Detokenize;
+use web3::contract::{Contract, Error};
+use web3::ethabi::Token;
+use web3::transports::Http;
+use web3::types::{BlockNumber, FilterBuilder, H256};
+use web3::Web3;
+
+use crate::node::client_handling::websocket::connection_handler::authenticated::RequestHandlingError;
+use credentials::token::bandwidth::TokenCredential;
+use crypto::asymmetric::identity::{PublicKey, Signature};
+use erc20_bridge_contract::msg::ExecuteMsg;
+use erc20_bridge_contract::payment::LinkPaymentData;
+use gateway_client::bandwidth::eth_contract;
+use network_defaults::{COSMOS_CONTRACT_ADDRESS, DENOM, ETH_EVENT_NAME};
+use validator_client::nymd::wallet::DirectSecp256k1HdWallet;
+use validator_client::nymd::{
+    AccountId, Denom, Fee, Gas, SigningCosmWasmClient, SigningNymdClient,
+};
+
+#[derive(Clone)]
+pub(crate) struct ERC20Bridge {
+    // This is needed because web3's Contract doesn't sufficiently expose it's eth interface
+    web3: Web3<Http>,
+    contract: Contract<Http>,
+    nymd_client: SigningNymdClient,
```

Comment:
To make your life easier you'd probably want a `NymdClient<SigningNymdClient>` here

Reply:
That simplified it a lot, thanks man.